packages/beacon-node/src/chain/validation/attestation.ts

```diff
@@ -65,6 +70,108 @@ export async function validateGossipAttestation(
   return validateAttestation(fork, chain, attestationOrBytes, subnet);
 }
 
+/**
+ * Verify gossip attestations of the same attestation data. The main advantage is we can batch verify bls signatures
+ * through verifySignatureSetsSameMessage bls api to improve performance.
+ *   - If there are less than 2 signatures (minSameMessageSignatureSetsToBatch), verify each signature individually with batchable = true
+ *   - do not prioritize bls signature set
+ */
+export async function validateGossipAttestationsSameAttData(
+  fork: ForkName,
+  chain: IBeaconChain,
+  attestationOrBytesArr: AttestationOrBytes[],
+  subnet: number,
+  // for unit test, consumers do not need to pass this
+  phase0ValidationFn = validateGossipAttestationNoSignatureCheck
```

Comment:
Unless this is talking about the phase0 fork, would recommend changing this language to "step 0", "step 1", etc.